codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -786,332 +786,338 @@ let checkAndFetchForChain = (
-          reorgChainRolledBackLastBlockData
-        } else {
-          //For all other chains, rollback to where a blockTimestamp is less than or equal to the block timestamp
-          //where the reorg chain is rolling back to
-          cf.lastBlockScannedHashes->ReorgDetection.LastBlockScannedHashes.rollBackToBlockTimestampLte(
-            ~blockTimestamp=lastKnownValidBlockTimestamp,
+    | Rollback =>
+      //If it isn't processing a batch currently continue with rollback otherwise wait for current batch to finish processing
+      switch state {
+      | {currentlyProcessingBatch: false, rollbackState: RollingBack(rollbackChain)} =>
+        Logging.warn("Executing rollback")
+        let chainFetcher = state.chainManager.chainFetchers->ChainMap.get(rollbackChain)
+        let rollbackChainId = rollbackChain->ChainMap.Chain.toChainId
+        //Get rollback block and timestamp
+        let reorgChainRolledBackLastBlockData =
+          await chainFetcher->rollbackLastBlockHashesToReorgLocation
+
+        let {blockNumber: lastKnownValidBlockNumber, blockTimestamp: lastKnownValidBlockTimestamp} =
+          reorgChainRolledBackLastBlockData->ChainFetcher.getLastScannedBlockData
+
+        let chainFetchers = state.chainManager.chainFetchers->ChainMap.mapWithKey((chain, cf) => {
+          let rolledBackLastBlockData = if chain == rollbackChain {
+            //For the chain fetcher of the chain where a  reorg occured, use the the
+            //rolledBackLastBlockData already computed
+            reorgChainRolledBackLastBlockData
+          } else {
+            //For all other chains, rollback to where a blockTimestamp is less than or equal to the block timestamp
+            //where the reorg chain is rolling back to
+            cf.lastBlockScannedHashes->ReorgDetection.LastBlockScannedHashes.rollBackToBlockTimestampLte(
+              ~blockTimestamp=lastKnownValidBlockTimestamp,
+            )
+          }
+
+          //Roll back chain fetcher with the given rolledBackLastBlockData
+          cf
+          ->ChainFetcher.rollbackToLastBlockHashes(~rolledBackLastBlockData)
+          ->ChainFetcher.addProcessingFilter(
+            ~filter=eventBatchQueueItem => {
+              let {timestamp, blockNumber} = eventBatchQueueItem
+              //Filter out events that occur passed the block where the query starts but
+              //are lower than the timestamp where we rolled back to
+              (timestamp, chain->ChainMap.Chain.toChainId, blockNumber) >
+              (lastKnownValidBlockTimestamp, rollbackChainId, lastKnownValidBlockNumber)
+            },
+            ~isValid=(~fetchState) => {
+              //Remove the event filter once the fetchState has fetched passed the
+              //timestamp of the valid rollback block's timestamp
+              let {blockTimestamp, blockNumber} = FetchState.getLatestFullyFetchedBlock(fetchState)
+              (blockTimestamp, chain->ChainMap.Chain.toChainId, blockNumber) <=
```

Comment:
I think the name of the `addProcessingFilter` and the fact that I get `chain` from the scope instead of the `~isValid` args are the only changes in the file